lib/rack/idempotency.rb

```diff
@@ -19,23 +20,47 @@ class Idempotency
     def initialize(app, store: NullStore.new)
       @app     = app
       @store   = store
+      if @store.class.to_s == "Rack::Idempotency::RedisStore"
```

Comment:
Not a good design: we introduce an unnecessary dependency on the exact name of the store class. This is Ruby, use duck typing:

```ruby
if store.respond_to?(:lock)
  @mutex_mode = true
end
```

Now no dependency, and also, anyone could implement locking mechanism on other store classes too.